src/app.ts

```diff
@@ -0,0 +1,33 @@
+import express from 'express';
+
+import ChatterService from './services/ChatterService';
+
+const app = express();
+app.use(express.json());
+
+let viewerwersPoint = new Map();
+
+setInterval(async () => {
+  try {
+    const viewers = await ChatterService.getViewers();
+
+    viewers.forEach((viewer) => {
+      let point = viewerwersPoint.get(viewer) ?? 0;
+      viewerwersPoint.set(viewer, ++point);
+      console.log(`${viewer} have ${point} points`);
+    });
+  } catch (error) {
+    console.log(error);
+  }
+}, 60 * 1000);
+
+app.get('/', (req, res) => {
+  const factoryViewer = ([name, points]) => ({ name, points });
+  let viewers = Array.from(viewerwersPoint.entries()).map(factoryViewer);
+  let top10 = viewers
+    .sort((a, b) => (a.points > b.points ? 1 : -1))
+    .slice(0, 10);
```

Comment:
Não sei o que é o Map() que o viewers é. Mas se for Immutable ele tem uma função sortBy que acho que deixaria o código mais curto. `viewers.sortBy(e => e.get('points'))`.
Nesse caso teria que usar o próprio viewerwersPoint e não o `entries()` porque aqui está transformando em Array mesmo né. Não entendi bem por que.

Reply:
Não é Immutable, porém o ```entries()``` realmente não apresenta alteração no valor de ```viewers``` que é ```<{name: string; points: number}[]>